docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+to sign out. This can occur where the first activity executed after signing
+out is the login activity; if this invokes the Smart Lock / OpenYOLO retrieve
+flows, this can result in automatically signing in the user again, without
+providing them any opportunity to intervene.
+
+This is prevented in Smart Lock for Passwords by invoking the
+`disableAutoSignIn` method:
+
+```java
+void signOut() {
+  Auth.CredentialsApi.disableAutoSignIn(mGoogleApiClient);
+  discardAuthToken();
+  goToLogin();
+}
+```
+
+The same method is provided in the OpenYOLO API:
+
+```java
+void signOut() {
+  mOpenYoloClient.disableAutoSignIn();
+  discardAuthToken();
+  goToLogin();
+}
+```
+
+OpenYOLO stores this state, by default, using a shared preferences file in the
+context of the client application. This behavior can be customized, to save
+this preference wherever you wish, by providing an alternative implementation
+of `AppSettings` via the `CredentialClientOptions` type, which can be provided
+to a `CredentialClient` during construction.
+
+Automatic sign-in can also be explicitly disabled in OpenYOLO at the start of
+the retrieve flow, by setting the `requireUserMediation` property to `true`:
+
+```java
+void sendOpenYoloRetrieveRequest() {
+  CredentialRetrieveRequest request = new CredentialRetrieveRequest.Builder(
+      AuthenticationMethods.EMAIL)
+      // ... other settings
+      setRequireUserMediation(true)
+      .build();
+
+  startActivityForResult(
+      mOpenYoloClient.getCredentialRetrieveIntent(request),
+      RC_RETRIEVE);
+}
+```
+
+## Deleting invalid credentials
+
+Occasionally, a credential held by the user's credential manager is found to
+be invalid during authentication. A client can request that this credential
+be deleted, in order to avoid the frustrating experience of a stale credential
+being offered to the user repeatedly.
+
+Smart Lock for Passwords follows the same two-stage pattern as retrieve and
+save for this functionality, and looks like the following:
+
+```java
+void deleteSmartLockCredential(Credential credential) {
+  Auth.CredentialsApi.delete(mGoogleApiClient, credential)
+      .setResultCallback(this::handleSmartLockDeleteResult);
+}
+
+void handleSmartLockDeleteResult(Status status) {
+  if (status.isSuccess()) {
+    // credential automatically deleted
+  } else if (status.hasResolution()) {
+    try {
+      status.startResolutionForResult(this, RC_DELETE);
+    } catch (IntentSender.SendIntentException e) {
+      Log.e(TAG, "Oh no :(");
+    }
+  } else {
+    // delete refused by provider or failed
+  }
+}
+
+@Override protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+  super.onActivityResult(requestCode, resultCode, data);
+  if (requestCode == RC_DELETE) {
+    if (resultCode == RESULT_OK) {
+      // credential deleted
+    } else {
+      // credential not deleted
+    }
+  }
+}
+```
+
+In OpenYOLO, an Intent is used, simplifying the code to the following:
```

Comment:
Same observation about Intent and backticks here

Reply:
Done.